### PR TITLE
[libc] Add definitions for totalordermag(,f, l, f128), dsqrt(l, f128), fsqrt(, l, f128) to math.yaml.

### PR DESCRIPTION
**Pull request by wldfngrs**
Added auto function hdrgen specification for functions: totalordermagf, dsqrtl, fsqrt, fsqrtl

### CODE REVIEW

**Comment by wldfngrs** [comment]
@lntue PR for review, thanks!

**Comment by jhuber6** [comment]
For now, we probably need to add these to `spec/stdc.td` as well since we're in-between formats. Also, do we have implementations for these?

File: libc/newhdrgen/yaml/math.yaml [hunk 1]
```diff
@@ -214,6 +214,12 @@ functions:
     arguments:
       - type: long double
       - type: long double
+  - name: dsqrtl
```
Comment (lntue) [inline libc/newhdrgen/yaml/math.yaml:217]: Can you also add the missing `dsqrtf128`?

File: libc/newhdrgen/yaml/math.yaml [hunk 2]
```diff
@@ -1195,6 +1201,18 @@ functions:
       - type: long double
       - type: int
       - type: unsigned int
+  - name: fsqrt
```
Comment (lntue) [inline libc/newhdrgen/yaml/math.yaml:1204]: Can you also add the missing `fsqrtf128`?

File: libc/newhdrgen/yaml/math.yaml [hunk 3]
```diff
@@ -2323,6 +2341,27 @@ functions:
...
+      - type: double *
+      - type: double *
+  - name: totalordermagf
+    standards:
+      - stdc
+    return_type: int
+    arguments:
+      - type: float *
+      - type: float *
+  - name: totalordermagl
```
Comment (lntue) [inline libc/newhdrgen/yaml/math.yaml:2358]: Can you also add the missing `totalordermagf128`?

**Comment by lntue** [comment]
> For now, we probably need to add these to `spec/stdc.td` as well since we're in-between formats. Also, do we have implementations for these?

Those functions were implemented and only added to `spec/stdc.td`, `spec/llvm_libc_ext.td`, but not `math.yaml`.

**Comment by wldfngrs** [comment]
@lntue Made a commit adding the function definitions suggested

File: libc/newhdrgen/yaml/math.yaml [hunk 4]
```diff
@@ -1195,13 +1209,32 @@ functions:
...
+    arguments:
+      - type: long double 
   - name: fsub
     standards:
       - stdc
     return_type: float
     arguments:
       - type: double
       - type: double
+  - name: fsqrtf128
```
Comment (lntue) [inline libc/newhdrgen/yaml/math.yaml:1231]: nit: put it together with other `fsqrt` functions.

**Review by lntue — APPROVED** [review]

File: libc/newhdrgen/yaml/math.yaml [hunk 5]
```diff
@@ -2323,6 +2356,35 @@ functions:
...
+      - type: long double *
+      - type: long double *
+  - name: totalordermagf128
+    standards:
+      - stdc
+    return_type: int
+    arguments:
+      - type: float128 *
+      - type: float128 *
+    guard: LIBC_TYPES_HAS_FLOAT128
```
Comment (overmighty) [inline libc/newhdrgen/yaml/math.yaml:2387]: `totalordermag*` functions take pointers to `const` values. See section F.10.12.2 of https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3096.pdf.

Comment (overmighty) [reply libc/newhdrgen/yaml/math.yaml:2387]: I opened #103935.

**Pull request merged**